From ae342cd6ce6418ececdf64f98feddb9a994f7730 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 11 Feb 2018 17:23:31 +0200
Subject: [PATCH 2240/2241] fix: net: mvpp2x: fix mv_pp2x_rxq_number
 calculation

Patch fix Netmap traffic stuck.
mv_pp2x_rxq_number calculation were wrongly based on
mv_pp2x_rx_count static value instead of num_active_cpus.

Number of RXQ in the system should be equal to
number of CoS * num_active_cpus and less than 32 (maximum physical
RXQs in packet processor HW).

Change-Id: I09e7a30591568e6d5d29dda638d60ec01b5db56f
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/50316
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index f242299..ab6588c 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -7052,7 +7052,7 @@ static int mv_pp2x_rxq_number_get(void)
 		if (mv_pp2x_num_cos_queues * num_active_cpus() > MVPP22_MAX_NUM_RXQ)
 			rx_queue_num = MVPP22_MAX_NUM_RXQ;
 		else
-			rx_queue_num = mv_pp2x_num_cos_queues * mv_pp2x_rx_count;
+			rx_queue_num = mv_pp2x_num_cos_queues * num_active_cpus();
 	else
 		rx_queue_num = mv_pp2x_num_cos_queues;
 
-- 
2.7.4

